hikari/impl/entity_factory.py

```diff
@@ -2383,13 +2429,22 @@ def serialize_command_permission(self, permission: commands.CommandPermission) -
         return {"id": str(permission.id), "type": permission.type, "permission": permission.has_access}
 
     def deserialize_partial_interaction(self, payload: data_binding.JSONObject) -> base_interactions.PartialInteraction:
+        authorizing_integration_owners = {
+            application_models.ApplicationIntegrationType(int(integration_type)): snowflakes.Snowflake(
+                int(integration_owner_id)
+            )
```

Comment:
I believe for snowflakes, you do not need to cast it to an int before hand (as Snowflake is a subclass of int)